core/shared/src/main/scala/laika/parse/code/languages/YAMLSyntax.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package laika.parse.code.languages
+
+import cats.data.NonEmptyList
+import laika.ast.CodeSpan
+import laika.bundle.SyntaxHighlighter
+import laika.parse.builders._
+import laika.parse.code.common.{Comment, Identifier, NumberLiteral, StringLiteral}
+import laika.parse.code.{CodeCategory, CodeSpanParser}
+import laika.parse.implicits._
+import laika.parse.text.TextParsers.{literal, oneOf}
+
+/**
+ * @author Michał Sitko
+ */
+object YAMLSyntax extends SyntaxHighlighter {
+  /** The names of the language (and its optional aliases) as used in text markup */
+  override def language: NonEmptyList[String] = NonEmptyList.of("yaml", "yml")
+
+  private val keyName = Identifier.alphaNum.withIdStartChars('_', '-', '.').withCategory(CodeCategory.AttributeName)
+
+  private val valueLiteral = {
+    val doubleQuotedStrLiteral = StringLiteral.singleLine('"')
+    val singleQuotedStrLiteral = StringLiteral.singleLine('\'')
+    val number = NumberLiteral.decimalFloat | NumberLiteral.decimalInt
+    val boolean = (literal("true") | literal("false")).map(b => Seq(CodeSpan(b, CodeCategory.BooleanLiteral)))
```

Comment:
YAML has a long list of valid boolean literals:
```scala
val boolean = Keywords(CodeCategory.BooleanLiteral)(
  "true","True","TRUE","false","False","FALSE","on","On","ON","off","Off","OFF","yes","Yes","YES","no","No","NO"
)
```

Reply:
It depends on YAML spec version, in 1.2 it's only 2 literals: https://github.com/go-yaml/yaml/issues/214

Anyway, I think I will go with 1.1 as you suggested, probably it's makes more sense to be more liberal for highlighting

Reply:
I wasn't aware of that change, I looked at an unversioned grammar document on yaml.org and tested a few common parsers like `pyyaml` which apparently never got updated for YAML 1.2.

I don't have a strong opinion either way, but if the 1.1 booleans are still in wide-spread use it might indeed make sense to support them.